go/vt/vtgate/semantics/info_schema.go

```diff
@@ -0,0 +1,973 @@
+	cols["SQL_MODE"] = query.Type(2075)
+	cols["DEFINER"] = query.Type(6165)
+	cols["CHARACTER_SET_CLIENT"] = query.Type(6165)
+	cols["COLLATION_CONNECTION"] = query.Type(6165)
+	cols["DATABASE_COLLATION"] = query.Type(6165)
+	infSchema["TRIGGERS"] = cols
+	cols = map[string]query.Type{}
+	cols["USER"] = query.Type(6167)
+	cols["HOST"] = query.Type(6167)
+	cols["ATTRIBUTE"] = query.Type(6163)
+	infSchema["USER_ATTRIBUTES"] = cols
+	cols = map[string]query.Type{}
+	cols["GRANTEE"] = query.Type(6165)
+	cols["TABLE_CATALOG"] = query.Type(6165)
+	cols["PRIVILEGE_TYPE"] = query.Type(6165)
+	cols["IS_GRANTABLE"] = query.Type(6165)
+	infSchema["USER_PRIVILEGES"] = cols
+	cols = map[string]query.Type{}
+	cols["TABLE_CATALOG"] = query.Type(6165)
+	cols["TABLE_SCHEMA"] = query.Type(6165)
+	cols["TABLE_NAME"] = query.Type(6165)
+	cols["SPECIFIC_CATALOG"] = query.Type(6165)
+	cols["SPECIFIC_SCHEMA"] = query.Type(6165)
+	cols["SPECIFIC_NAME"] = query.Type(6165)
+	infSchema["VIEW_ROUTINE_USAGE"] = cols
+	cols = map[string]query.Type{}
+	cols["VIEW_CATALOG"] = query.Type(6165)
+	cols["VIEW_SCHEMA"] = query.Type(6165)
+	cols["VIEW_NAME"] = query.Type(6165)
+	cols["TABLE_CATALOG"] = query.Type(6165)
+	cols["TABLE_SCHEMA"] = query.Type(6165)
+	cols["TABLE_NAME"] = query.Type(6165)
+	infSchema["VIEW_TABLE_USAGE"] = cols
+	cols = map[string]query.Type{}
+	cols["TABLE_CATALOG"] = query.Type(6165)
+	cols["TABLE_SCHEMA"] = query.Type(6165)
+	cols["TABLE_NAME"] = query.Type(6165)
+	cols["VIEW_DEFINITION"] = query.Type(6163)
+	cols["CHECK_OPTION"] = query.Type(2074)
+	cols["IS_UPDATABLE"] = query.Type(2074)
+	cols["DEFINER"] = query.Type(6165)
+	cols["SECURITY_TYPE"] = query.Type(6165)
+	cols["CHARACTER_SET_CLIENT"] = query.Type(6165)
+	cols["COLLATION_CONNECTION"] = query.Type(6165)
+	infSchema["VIEWS"] = cols
+	return infSchema
+}
+
+type infoSchemaWithColumns struct {
+	inner          SchemaInformation
+	infoSchemaData map[string]map[string]query.Type
+}
+
+// newSchemaInfo returns a SchemaInformation that has the column information for all info_schema tables
+func newSchemaInfo(inner SchemaInformation) SchemaInformation {
+	return &infoSchemaWithColumns{inner: inner, infoSchemaData: getInfoSchema()}
+}
+
+// FindTableOrVindex implements the SchemaInformation interface
+func (i *infoSchemaWithColumns) FindTableOrVindex(tbl sqlparser.TableName) (*vindexes.Table, vindexes.Vindex, string, topodatapb.TabletType, key.Destination, error) {
+	if !strings.EqualFold(tbl.Qualifier.String(), "information_schema") {
+		return i.inner.FindTableOrVindex(tbl)
+	}
+
+	ks := vindexes.Keyspace{
+		Name:    "information_schema",
+		Sharded: false,
+	}
+	var cols []vindexes.Column
+	for name, typ := range i.infoSchemaData[strings.ToUpper(tbl.Name.String())] {
+		cols = append(cols, vindexes.Column{
+			Name: sqlparser.NewIdentifierCI(name),
+			Type: typ,
+		})
+	}
+
+	sort.Slice(cols, func(i, j int) bool {
+		return cols[i].Name.String() < cols[j].Name.String()
+	})
```

Comment:
should the columns be sorted? according to MySQL docs, they do not seem to follow a specific alphabetical order: https://dev.mysql.com/doc/refman/8.0/en/information-schema-administrable-role-authorizations-table.html

Reply:
I sort them because otherwise all the expandStar tests become flaky

Reply:
makes sense okay, should they be sorted the same way MySQL sort them though? I am unsure if this is relevant. But imagine someone doing a `select * from information_schema.COLUMNS` and they expect to have `TABLE_CATALOG ` as the first column, but instead our first column is `COLUMN_COMMENT `.

Reply:
fixed in my last commit